lib/mspec/expectations/should.rb

```diff
@@ -23,6 +23,10 @@ def should_not(matcher = NO_MATCHER_GIVEN)
     raise "should_not outside example" unless state
     MSpec.actions :expectation, state
 
+    if RaiseErrorMatcher === matcher
+      $stderr.puts "\nDeprecation: ->{}.should_not raise_error breaks code style and is deprecated"
```

Comment:
We should show the caller here, so it's easy to know where it comes from.

Reply:
Could you use MSpec.deprecate (https://github.com/ruby/mspec/blob/47e3cbf134b7b0e1b425a607bfa1d213e9974d3e/lib/mspec/utils/deprecate.rb#L2)?
That will automatically include the caller.

Reply:
`replacement` can be e.g. "a matcher to verify the result"

Reply:
Yeah, will try. I saw this helper but decided not to use it because has a little bit different purpose - to notify about changing API or DSL. But here we have a code style issue.

Nevertheless it should work well for our case.

Reply:
Done